htdocs/compta/bank/transfer.php

```diff
@@ -189,10 +189,10 @@
 					$error++;
 				}
 				if (!$error) {
-					$mesgs = $langs->trans("TransferFromToDone", '{s1}', '{s2}', $amount[$n], $langs->transnoentitiesnoconv("Currency".$conf->currency));
-					$mesgs = str_replace('{s1}', '<a href="bankentries_list.php?id='.$tmpaccountfrom->id.'&sortfield=b.datev,b.dateo,b.rowid&sortorder=desc">'.$tmpaccountfrom->label.'</a>', $mesgs);
-					$mesgs = str_replace('{s2}', '<a href="bankentries_list.php?id='.$tmpaccountto->id.'">'.$tmpaccountto->label.'</a>', $mesgs);
-					setEventMessages($mesgs, null, 'mesgs');
+					$mesg = $langs->trans("TransferFromToDone", '{s1}', '{s2}', $amount[$n], $langs->transnoentitiesnoconv("Currency".$conf->currency));
+					$mesg = str_replace('{s1}', '<a href="bankentries_list.php?id='.$tmpaccountfrom->id.'&sortfield=b.datev,b.dateo,b.rowid&sortorder=desc">'.$tmpaccountfrom->label.'</a>', $mesg);
+					$mesg = str_replace('{s2}', '<a href="bankentries_list.php?id='.$tmpaccountto->id.'">'.$tmpaccountto->label.'</a>', $mesg);
+					setEventMessages($mesg, null, 'mesg');
```

Comment:
![image](https://github.com/Dolibarr/dolibarr/assets/3624836/1a349f02-0eac-4009-9859-ce1ba2e38f64)

Reply:
1. The first argument as defined in the the method definition of setEventMessages is "$mesg".  Phan triggers when the variable provided as the argument in the function call is $mesgs which is the second argument of the method is "mesgs".  So it flags a suspicious arguments order or position.
2. I try to modify as little as possible - possibly setEventMessages($mesg) is ok, but the default style is 'mesgs', and the original call requests 'mesg' as the style.

Reply:
did you try setEventMessage without 's', default style is mesgs

Reply:
Ok, sorry, I see the issue now, my replacement also replaced the third parameter to 'mesg' while it was 'mesgs' which is the default.

So I further simplified the statement now as you suggested.